### DISCUSSION TITLE
Running with Docker under NGINX Proxy results in malformed webpage

### DISCUSSION THREAD

**Discussion by hwittenborn** [dani-garcia/vaultwarden#1383] [Q&A] [answered]
### Subject of the issue

As per the title, I'm trying to run the docker image for bitwarden_rs, with request being brought from an NGINX reverse proxy. When accessing the page from the proxy, several details are missing, but if I do a direct connect to the container via it's published port, it looks just fine.
### Your environment

* Bitwarden_rs version: 1.19.0

* Install method: Docker image
* Clients used:  Web/Firefox
* Reverse proxy and version:  NGINX 1.18.0
* Version of mysql/postgresql:  N/A
* Other relevant information:

**Images:**
NGINX: 
![Screenshot from 2021-02-10 23-29-30](https://user-images.githubusercontent.com/74838472/107605493-ea2fe180-6bf8-11eb-8e1a-744358b58b40.png)

Directly to container:
![Screenshot from 2021-02-10 23-29-28](https://user-images.githubusercontent.com/74838472/107605512-f9af2a80-6bf8-11eb-8b74-b6a5ae07549a.png)
--
``` 
**docker-compose.yml**
version: "3.5"
services:
  bitwarden:
    container_name: bitwarden
    image: bitwardenrs/server:latest
    restart: unless-stopped
    networks:
      - bitwarden
    ports:
      - REDACTED:80
    volumes:
      - REDACTED:/data
    environment:
      DOMAIN: 'REDACTED'
      SIGNUPS_ALLOWED: 'false'
      INVITATIONS_ALLOWED: 'false'
networks:
  bitwarden:
    name: REDACTED

**NGINX**
server {
  server_name REDACTED;
  location / {
    proxy_pass http://localhost:REDACTED;
    }
*continues with default certbot config* ```

**Comment by reinob** [comment]
Your reverse proxy config looks incomplete.
Why don't use use the one in https://github.com/dani-garcia/bitwarden_rs/wiki/Proxy-examples (Nginx, by shauder).
This works just fine on my Raspberry Pi 4 (running debian).

**Comment by hwittenborn** [comment]
I added the following to `location /`, still no luck --
`    proxy_set_header X-Real-IP $remote_addr;`
`    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;`
`    proxy_set_header X-Forwarded-Proto $scheme;`

I also added this part; again, no luck.
`  location /notifications/hub/negotiate {`
`    proxy_pass http://<SERVER>:80;`
`  }`

I've omitted everything else as it doesn't seem to be of any value to me.

I do remember it working before with the "minimal config" I had set - but when I changed some DNS settings in Cloudflare(changed it from proxying to direct connection) all these issues started popping up. For some reason though this kind of issue is only happening with bitwarden_rs and nothing else I'm running.

**Comment by hwittenborn** [comment]
Checking a few things now, and it appears the issue is happening when returning to Cloudflare as well - I guess something was just cached or something, and thus hiding the problems from me for a while.

**Comment by hwittenborn** [comment]
Some more stuff:
The admin page is also all janky - it's like something isn't loading or something. All I get is HTTP 200 codes in the Firefox Dev Console, but I'm also noticing the following directories aren't available through NGINX(but are through a direct connection):

`/app/vendor.db46b82e007a606bb83b.js`
`/app/main.db46b82e007a606bb83b.js`

Still can't think of why in all of this though

**Comment by hwittenborn** [comment, marked as answer]
Just tried something 'cause I didn't really know what else - but reinstalling NGINX fixed the issue for some reason.

Anyhow, I'm all good now. Don't know why it worked, but it did

**Comment by MeiRos** [comment]
Edit2 *no, it's not working. Only sometimes... there are now upstream timed out errors...
With browser, it will not load passwords*
Edit1 *it's working now. Not sure why. I did 'cp mysite.conf default.conf'. Tested and it worked. 'rm default.conf' and still working.*

I just updated and now I have the same problem.
With mobile browser I get only Bitwarden logo. Didn't try pc. https://mydomain/images/logo-dark@2x.png
Admin page shows normal text. (Not enabled and so on)

I'm using raspberry pi and nginx proxy. Nginx is docker.

What should I do? Doing 'sudo docker-compose pull' (+down and up commands too) again didn't help. How to reinstall it?

This is the start of log. Sudo docker-compose logs after up.

```
nginx-proxy    | /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
nginx-proxy    | /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
nginx-proxy    | /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
nginx-proxy    | 10-listen-on-ipv6-by-default.sh: info: /etc/nginx/conf.d/default.conf is not a file or does not exist
nginx-proxy    | /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
bitwarden_1    | /--------------------------------------------------------------------\
nginx-proxy    | /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
bitwarden_1    | |                       Starting Bitwarden_RS                        |
nginx-proxy    | /docker-entrypoint.sh: Configuration complete; ready for start up
bitwarden_1    | |                           Version 1.19.0
```
